templates/common/_base/files/configure-ovs-network.yaml

```diff
@@ -495,6 +614,41 @@ contents:
         fi
       }
 
+      ovnk_config_dir='/etc/ovnk'
+      extra_bridge_file="${ovnk_config_dir}/extra_bridge"
+      iface_default_hint_file="${ovnk_config_dir}/iface_default_hint"
+
+      # make sure to create ovnk_config_dir if it does not exist, yet
+      if ! [ -d "${ovnk_config_dir}" ]; then
+        mkdir -p "${ovnk_config_dir}"
+      fi
+
+      # For upgrade scenarios, make sure that we stabilize what we already configured
+      # before. If we do not have a valid interface hint, find the physical interface
+      # that's attached to ovs-if-phys0.
+      # If we find such an interface, write it to the hint file.
+      iface_default_hint=$(get_iface_default_hint "${iface_default_hint_file}")
+      if [ "${iface_default_hint}" == "" ]; then
+        current_interface=$(get_bridge_physical_interface ovs-if-phys0)
+        if [ "${current_interface}" != "" ]; then
+          write_iface_default_hint "${iface_default_hint_file}" "${current_interface}"
+        fi
+      fi
+
+      # delete iface_default_hint_file if it has the same content as extra_bridge_file
+      # in that case, we must also force a reconfiguration of our network interfaces
+      # to make sure that we reconcile this conflict
+      if [ -f "${iface_default_hint_file}" ] &&
+         [ -f "${extra_bridge_file}" ] &&
+         [ "$(cat "${iface_default_hint_file}")" == "$(cat "${extra_bridge_file}")" ]; then
+        echo "${iface_default_hint_file} and ${extra_bridge_file} share the same content"
+        echo "Deleting file ${iface_default_hint_file} to choose a different interface"
+        rm -f "${iface_default_hint_file}"
+        if [ -f /run/configure-ovs-boot-done ]; then
+          rm -f /run/configure-ovs-boot-done
+        fi
+      fi
```

Comment:
How can we end up with the same content in iface_default_hint_file and extra_bridge_file? Are you contemplating for external manipulation of iface_default_hint_file or that someone changes extra_bridge_file to what is in iface_default_hint_file? I don't think we were currently handling elsewhere the case of extra_bridge_file being set to the interface the we had picked as default interface, is that it?

---

```
if [ -f /run/configure-ovs-boot-done ]; then
          rm -f /run/configure-ovs-boot-done
fi
```
could be just `rm -f /run/configure-ovs-boot-done`?

Reply:
I don't have a strong opinion here. We can kick the entire if condition out, I had thought of it as a kind of safety verification, though. tbh, I am manually running the script on a rhel 8 vm and I was just trying what would happen in case of misconfiguration.

> Are you contemplating for external manipulation of iface_default_hint_file or that someone changes extra_bridge_file
Yeah, I had actually thought of both. More so though the second option where someone might misconfigure extra_bridge_file

That's also why I changed the get_default_interface logic a bit, to:
~~~
471     get_default_interface() {
472       local iface=""
473       local counter=0
474       local iface_default_hint_file="$1"
475       local extra_bridge_file="$2"
476       local extra_bridge=""
477 
478       if [ -f "${extra_bridge_file}" ]; then
479         extra_bridge=$(cat ${extra_bridge_file})
480       fi
481 
482       # find default interface
483       # the default interface might be br-ex, so check this before looking at the hint
484       while [ ${counter} -lt 12 ]; do
485         # check ipv4
486         # never use the interface that's specified in extra_bridge_file
487         # never use br-ex1
488         if [ "${extra_bridge}" != "" ]; then
489           iface=$(ip route show default | grep -v "br-ex1" | grep -v "${extra_bridge}" | awk '{ if ($4 == "dev") { print $5; exit } }')
490         else
491           iface=$(ip route show default | grep -v "br-ex1" | awk '{ if ($4 == "dev") { print $5; exit } }')
492         fi
493         if [[ -n "${iface}" ]]; then
494           break
495         fi
496         # check ipv6
497         # never use the interface that's specified in extra_bridge_file
498         # never use br-ex1
499         if [ "${extra_bridge}" != "" ]; then
500           iface=$(ip -6 route show default | grep -v "br-ex1" | grep -v "${extra_bridge}" | awk '{ if ($4 == "dev") { print $5; exit } }')
501         else
502           iface=$(ip -6 route show default | grep -v "br-ex1" | awk '{ if ($4 == "dev") { print $5; exit } }')
503         fi
~~~